hbase_sql/src/main/java/com/splicemachine/derby/stream/spark/SparkDataSetProcessor.java

```diff
@@ -306,6 +313,27 @@ public Partitioner getPartitioner(DataSet<LocatedRow> dataSet, ExecRow template,
         }
     }
 
+    /**
+     *
+     * Spark cannot handle empty directories.  Unfortunately, sometimes tables are empty.  Returns
+     * empty when it can infer the error and check the directory for files.
+     *
+     * @param e
+     * @param <V>
+     * @return
+     * @throws Exception
+     */
+    private <V> DataSet<V> handleExceptionInCaseOfEmptySet(Exception e, String location) throws Exception {
+        // Cannot Infer Schema, Argh
+        if (e instanceof AnalysisException && e.getMessage() != null && e.getMessage().startsWith("Unable to infer schema")) {
+            // Lets check if there are existing files...
+            FileInfo fileInfo = ImportUtils.getImportFileInfo(location);
+            if (fileInfo.fileCount() == 0) // Handle Empty Directory
+                return getEmpty();
+        }
+        throw e;
+    }
+
```

Comment:
I figured an empty table would be a special case.  I do not want to recurse the files unless I have to (exception).